families/supplychain/python/sawtooth_supplychain/rest_api/route_handlers.py

```diff
@@ -0,0 +1,795 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+# Supply Chain REST API
+
+import re
+import logging
+import json
+from concurrent.futures import ThreadPoolExecutor
+import psycopg2
+
+from aiohttp import web
+
+import supplychain.python.sawtooth_supplychain.rest_api.exceptions as errors
+
+
+DEFAULT_TIMEOUT = 300
+LOGGER = logging.getLogger(__name__)
+
+POSTGRES_CONNECTION_STRING = \
+    "dbname='mydb' user='testuser' host='localhost' password='password'"
+
+
+class RouteHandler(object):
+    """Contains a number of aiohttp handlers for endpoints in
+    the Supply Chain Rest Api.
+
+    Each handler takes an aiohttp Request object, and uses the data in
+    that request to create queries to send to the database. The query
+    results are processed, and finally an aiohttp Response object is
+    sent back to the client with JSON formatted data and metadata.
+
+    If something goes wrong, an aiohttp HTTP exception is raised or returned
+    instead.
+
+    Args:
+        stream (:obj: messaging.stream.Stream):
+        timeout (int, optional): The time in seconds before the Api should
+            cancel a request and report that the validator is unavailable.
+    """
+    def __init__(self, loop, stream, timeout=DEFAULT_TIMEOUT):
+        loop.set_default_executor(ThreadPoolExecutor())
+        self._loop = loop
+        self._stream = stream
+        self._timeout = timeout
+
+    @staticmethod
+    def _db_connect():
+        """Connect to a PostgreSQL database. The credentials are currently
+        included in the hard-coded connection string, but a more sophisticated
+        method could be implemented if needed.
+        """
+        try:
+            conn = psycopg2.connect(POSTGRES_CONNECTION_STRING)
+        except psycopg2.OperationalError as e:
+            raise e
+        return conn
+
+    @staticmethod
+    def _db_close(conn, cur):
+        """Closes the open PostgreSQL connection.
+        """
+        conn.commit()
+        cur.close()
+        conn.close()
+
+    @staticmethod
+    def _check_paging_params(count, p_min, p_max):
+        """Check paging params to make sure they are integers and
+        that they are not out of range.
+        """
+        try:
+            count = int(count)
+            p_min = int(p_min)
+            if p_max is not None:
+                p_max = int(p_max)
+        except:
+            LOGGER.debug("Non-integer paging parameter.")
+            raise errors.InvalidPagingQuery()
+
+        if int(count) > 1000:
```

Comment:
This is a magic number, littered everywhere.  Make this a constant, `DEFAULT_PAGE_SIZE`